:sparkles: #2 - Add validation.

svenvandescheur opened the pull request:
Closes #2 

TODO:

- [x] Stories
- [x] Pass validators
- [x] Discuss: https://github.com/open-formulieren/formio-renderer/pull/25/files#diff-9185d63b2c63822bf756d0c5012592387413b1089b5ea674d96d922e159e4e15R153

sergei-maertens reviewed (changes requested):
Main principles look fine, just a couple remarks and thoughts

svenvandescheur commented on src/lib/renderer/renderer.stories.tsx:
```diff
@@ -35,10 +36,81 @@ renderForm.args = {
+  expect(await canvas.findByText('Er zijn te weinig karakters opgegeven.'));
+
+  await userEvent.type(canvas.getByLabelText(FORMIO_PATTERN.label), '123A', {delay: 30});
+  expect(await canvas.findByText('De opgegeven waarde voldoet niet aan het formaat.'));
```
As discussed: `findByText` is a more effective approach (without empty expect).

svenvandescheur commented:
@sergei-maertens I've replaced the `getComponentByKey` function with the Formik utils version, and reverted that change as it broke the build (build doesn't complete after this change).

sergei-maertens commented on src/lib/validation/validate.ts:
```diff
@@ -0,0 +1,171 @@
+    await validateForm(form, values, validators);
+    return;
+  } catch (result) {
+    // Convert the validation errors to messages.
+    const entries = Object.entries(result).map(
+      ([key, validationErrors]: [string, ValidationError[]]) => {
+        const messages = validationErrors
+          .map(validationError => validationError.message.trim())
+          .join('\n');
+        return [key, messages];
+      }
+    );
```
I see - and at that point it might actually be more beneficial to do something like `errors as any as string` just to make TS shut up

sergei-maertens commented:
Merging this and I'll create a new PR to sort out the package build with `tsc` instead of this microbundle stuff
